Sources/Display.php

```diff
@@ -44,7 +44,7 @@ function Display()
 	if (isset($_SERVER['HTTP_X_MOZ']) && $_SERVER['HTTP_X_MOZ'] == 'prefetch')
 	{
 		ob_end_clean();
-		header('HTTP/1.1 403 Prefetch Forbidden');
+		send_http_status(403, 'Prefetch Forbidden');
```

Comment:
To use send_http_status is in my eyes not the right approache,
because this function is not everywhere defined.

Reply:
send_http_status is defined in Errors.php which loaded on every page load (wether from /index.php or /SSI.php). It's also available before any SMF functions are called so it is defined everywhere it's being used.

Reply:
nope see proxy.php
Why you choose to take this risk instead of place the vars ther?

Reply:
Since we use this on many early stages and error cases etc.
i would be for to keep the decentralize concept and use the var methode instead of a function approache.

Reply:
It's possible for SERVER_PROTOCOL to hold a few different values according to RFC 3875 and we need to default to HTTP/1.0 if we don't know the protocol. So the following is not guaranteed to work:
```header($_SERVER['SERVER_PROTOCOL'] . " 500 Internal Server Error");```

So we need to check the protocol using a preg_match which will look like this each time (if we assume that there could be excess whitespace on the protocol):
```header((preg_match('~^\s*(HTTP/[12]\.\d)\s*$~i', $_SERVER['SERVER_PROTOCOL'], $matches) ? $matches[1] : "HTTP/1.0") . " 500 Internal Server Error");```
Or
```header(preg_match('~^(HTTP/[12]\.\d)$~i', trim($_SERVER['SERVER_PROTOCOL'])) ? $_SERVER['SERVER_PROTOCOL'] : "HTTP/1.0") . " 500 Internal Server Error");```

Compared with my proposed version:
```send_http_status(500);```

With the function in Subs.php it is available in all cases before any SMF function calls. If defined there then it is *definitely* loaded whether in early stages or in an error cases.